parlai/crowdsourcing/utils/tests.py

```diff
@@ -119,3 +120,39 @@ def _register_mock_agents(self, num_agents: int = 1) -> List[str]:
         agent_ids = [agent.db_id for agent in agents]
 
         return agent_ids
+
+
+class AbstractOneTurnCrowdsourcingTest(AbstractCrowdsourcingTest):
+    """
+    Abstract class for end-to-end tests of one-turn crowdsourcing tasks.
+
+    Useful for Blueprints such as AcuteEvalBlueprint and StaticReactBlueprint for which
+    all of the worker's responses are sent to the backend code at once.
+    """
+
+    def _test_agent_state(self, expected_state: Dict[str, Any]):
+        """
+        Test that the actual agent state matches the expected state.
+
+        Register a mock human agent, request initial data to define the 'inputs' field
+        of the agent state, make the agent act to define the 'outputs' field of the
+        agent state, and then check that the agent state matches the desired agent
+        state.
+        """
+
+        # Set up the mock human agent
+        agent_id = self._register_mock_agents(num_agents=1)[0]
+
+        # Set initial data
+        self.server.request_init_data(agent_id)
+
+        # Make agent act
+        self.server.send_agent_act(
+            agent_id,
+            {"MEPHISTO_is_submit": True, "task_data": expected_state['outputs']},
```

Comment:
very nit: would this test break if my ouputs have any time-related values such as timestamp? is the "outputs" where task_start task_end field logged there?

Reply:
Hmmm, yes, if you have a timestamp in your output, this would break - lemme know if that's a common use case for you and we can find a workaround. The task_start and task_end fields get logged outside "outputs", in their own fields